src/MissionManager/MissionController.cc

```diff
@@ -256,6 +257,49 @@ bool MissionController::_convertToMissionItems(QmlObjectListModel* visualMission
     return endActionSet;
 }
 
+void MissionController::convertToKMLDocument(QDomDocument& document)
+{
+    QJsonObject missionJson;
+    QmlObjectListModel* visualItems = new QmlObjectListModel();
+    QList<MissionItem*> missionItens;
+    QString error;
+    save(missionJson);
+    _loadItemsFromJson(missionJson, visualItems, error);
+    _convertToMissionItems(visualItems, missionItens, this);
+
+    float altitude = missionJson[_jsonPlannedHomePositionKey].toArray()[2].toDouble();
+
+    QString coord;
+    QStringList coords;
+    for(const auto& item : missionItens) {
+        switch(item->command()) {
```

Comment:
This code should be change to use the MissionCommonUIInfo data which has meta data for each mission command:
```
    const MissionCommandUIInfo* uiInfo = qgcApp()->toolbox()->missionCommandTree()->getUIInfo(_controllerVehicle, item->command());
    if (uiInfo && uiInfo->specifiesCoordinate() && !uiInfo->standaloneCoordinate())  {
        // Add KML for a mission item which specifies a coordinate that the vehicle flies through
    }
```

Reply:
@DonLakeFlyer Done.